docs/src/main/sphinx/connector/hive.rst

```diff
@@ -247,9 +247,9 @@ Hive connector documentation.
               in schemas ``fruit`` and ``vegetable``
             * ``*`` to cache listings for all tables in all schemas
       -
-    * - ``hive.file-status-cache-size``
-      - Maximum total number of cached file status entries.
-      - 1,000,000
+    * - ``hive.file-status-cache-data-size``
+      - Maximum retained size of cached file status entries.
+      - ``1GB``
```

Comment:
Sounds like a lot.
Is it a safe default?

Reply:
It's disabled by default (`hive.file-status-cache-tables` is empty). 

Per transaction is 100MB by default (per query), so with 20 concurrency and heavy file listing it could top-out at 2GB. I assume it's fine because even for small coordinators (e.g 32GB), that's less than 10% of mem usage.

Reply:
> for small coordinators (e.g 32GB)

my assumption is that small coordinators are nowhere near 32 GB heap

i've been hearing different "stories" around minimal heap size over time. "trino isn't design for smartwatches"  "trino isn't supposed to run on mobile phones"... i don't think things like that should be based on anecdotal evidence though, so will try not to pay attention to these. If 32 is minimum, let's update https://trino.io/docs/current/installation/deployment.html to indicate that. It currently uses default of 16 GB and doesn't define any minimum.

> that's less than 10% of mem usage.

that would be fine assuming before the change coordinator JVM had 10% spare memory
adding 10% more memory usage is a big change.

can we have defaults that are more conservative and leave tuning specific to particular deployments where operator will know how much spare memory they have?

Reply:
> 've been hearing different "stories" around minimal heap size over time. "trino isn't design for smartwatches" "trino isn't supposed to run on mobile phones".

I think it relates mostly to workers where data processing velocity is much higher. However, coordinator also shouldn't be too small (mostly because of CPU usage).

> can we have defaults that are more conservative and leave tuning specific to particular deployments where operator will know how much spare memory they have?

Which setting you mean here? What do you suggest?

Reply:
`hive.file-status-cache.max-retained-size` is being documented to have 1 GB default

can it be off by default, or eg 10 MB?
also, is it safe to have this cache enabled by default? do we detect writes done by other applications/clusters?


`hive.per-transaction-file-status-cache.max-retained-size` is being documented to have 100 MB by default. with 20 concurrency (arbitrary number, since there isn't default concurrency limit, right?) this means up to 2GB. would it be safe to have eg 10 MB default? or 1 MB?